src/dune_lang/slang.ml

```diff
@@ -217,6 +222,11 @@ let rec simplify = function
 
 and simplify_blang = function
   | Blang.Const b -> Blang.Const b
+  | Expr (Literal s) ->
+    (match String_with_vars.text_only s with
+     | Some "true" -> Const true
+     | Some "false" -> Const false
+     | _ -> Expr (Literal s))
```

Comment:
Maybe bind the `Expr (Literal s)` in the match clause and reuse it here to show that it is the same. Potentially saves an allocation, but mainly a bit simpler to grasp that it doesn't simplify in that case.